### DISCUSSION TITLE
组件文档编写规范

### DISCUSSION THREAD

**Discussion by xingyan95** [DevCloudFE/vue-devui#119] [General]
### API表格
1. 所有`API`表格格式应该统一。推荐左对齐。
2. API表格的标题等级应该为三级，方便生成右侧【快速前往】目录。
3. API表格的标题，组成格式为：`d-xxx [参数 | 事件 | 方法 | 插槽]`，且组件名字后面加一个空格；类型定义放在最后，格式为`xx 类型定义`，类型名字后面同样需要加一个空格。
4. 【何时使用】标题不应该放入【快速前往】目录。
5. API表格中需要全部包含【跳转Demo】，【全局配置项】需要移除。
6. API表格的表头统一采用中文。
7. 参数类型需准确，例如枚举值的参数类型不可为`string`，应该写具体类型名字（如`TriggerType`），并且参考第3条的格式写明类型定义。
8. 【跳转Demo】链接需能够正确跳转，并且demo中需要展示对应API的用法。
9. 组件用到的默认插槽也需要做说明，并且插槽名字应该为`default`。
10. API参数基本类型采用首字母大写还是小写？`String` or `string`，自定义类型名字采用大驼峰命名，（如`TriggerType`）。
11. 每个参数需说明可选还是必选。

### 演示demo
1. 演示demo需要对所使用的API及组件默认行为和样式等进行详细的说明。避免让用户自己去推测，降低用户学习和使用成本。
2. 每个组件的第一个demo，应该是组件最基本的用法，即展示组件在不传参数或者传入最基本参数情况下的效果。以此引申出组件demo的展示顺序如何组织？比如第一个demo是最基本用法，后面的demo展示在第一个demo基础上新增一个或几个API的效果？
3. 每个demo所展示的API应该尽量精简，尽量不要一个demo中展示多个API的用法。
4. 当API是枚举值时，demo中应尽量展示每个枚举值的效果。
5. 演示demo代码，能换行的换行展示，避免出现横向滚动条。
6. 演示demo自定义class样式，命名时最好有前缀区分，不然会成为全局样式，有可能影响其他组件或者demo效果。
7. 文案描述需清晰，标点符号使用需正确。
8. 一个标题展示一个demo，避免一个标题中展示多个demo。
9. `boolean`类型的参数，在demo中展示，设置为`true`的时候，不需要显式的设置为`true`，直接写参数名字即可。
10. demo的描述说明文案中，用到代码块的地方需要用反引号包裹。
11. demo的描述说明文案应该跟在`:::demo`后面，不应该放在`h4`或其他标签中，也不应该放在代码块外面。

**Comment by kagol** [comment]
开发规范以贡献指南中的开发规范为准：
[https://vue-devui.github.io/contributing/](https://vue-devui.github.io/contributing/)
